internal/watcher/watcher_plugin.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+package watcher
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/nginx/agent/v3/internal/bus"
+	"github.com/nginx/agent/v3/internal/config"
+)
+
+type (
+	Watcher struct {
+		messagePipe            bus.MessagePipeInterface
+		agentConfig            *config.Config
+		instanceWatcherService *InstanceWatcherService
+		instanceUpdatesChannel chan InstanceUpdates
+		cancel                 context.CancelFunc
+	}
+)
+
+func NewWatcher(agentConfig *config.Config) *Watcher {
+	return &Watcher{
+		agentConfig:            agentConfig,
+		instanceWatcherService: NewInstanceWatcherService(agentConfig),
+		instanceUpdatesChannel: make(chan InstanceUpdates),
+	}
+}
+
+// nolint: unparam
+// error is always nil
+func (w *Watcher) Init(ctx context.Context, messagePipe bus.MessagePipeInterface) error {
+	slog.DebugContext(ctx, "Starting watcher plugin")
+	w.messagePipe = messagePipe
+
+	watcherContext, cancel := context.WithCancel(ctx)
+	w.cancel = cancel
+
+	go w.instanceWatcherService.Watch(watcherContext, w.instanceUpdatesChannel)
+	go w.monitorInstanceUpdates(watcherContext)
+
```

Comment:
Hmm, this is explicitly called out as a thing **not** to do in the Uber Style guide:
https://github.com/uber-go/guide/blob/master/style.md#no-goroutines-in-init
I suppose our hands are a little tied when it comes to the interface we have to implement.

Which reminds me, we should also verify interface compliance here too (verify that this is a Plugin):
https://github.com/uber-go/guide/blob/master/style.md#verify-interface-compliance

Reply:
This `Init` function here is different from the built in `init` function in golang so it is ok.
I'll add the interface compliance.

Reply:
Agree about the Init thing, think it's fine for us here 👌